packages/block-library/src/heading/variations.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * WordPress dependencies
+ */
+import { sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import HeadingLevelIcon from './heading-level-icon';
+
+const variations = [ 1, 2, 3, 4, 5, 6 ].map( ( level ) => ( {
+	name: `heading-${ level }`,
+	/* translators: %d: heading level. */
+	title: sprintf( 'Heading %d', level ),
+	icon: <HeadingLevelIcon level={ level } />,
```

Comment:
Aside: @gwwar, if we had better handling for icons, we could keep put those variations in `block.json`. Well, JS offers loops that make encoding those variations simpler. The downside is that variations aren't exposed on the server.